aws/rust-runtime/aws-runtime/src/invocation_id.rs

```diff
@@ -72,24 +73,26 @@ impl InvocationId {
 mod tests {
     use crate::invocation_id::InvocationIdInterceptor;
     use aws_smithy_http::body::SdkBody;
+    use aws_smithy_runtime_api::client::interceptors::context::phase::BeforeTransmit;
     use aws_smithy_runtime_api::client::interceptors::{Interceptor, InterceptorContext};
     use aws_smithy_runtime_api::config_bag::ConfigBag;
     use aws_smithy_runtime_api::type_erasure::TypedBox;
     use http::HeaderValue;
 
-    fn expect_header<'a>(context: &'a InterceptorContext, header_name: &str) -> &'a HeaderValue {
-        context
-            .request()
-            .unwrap()
-            .headers()
-            .get(header_name)
-            .unwrap()
+    fn expect_header<'a>(
+        context: &'a InterceptorContext<BeforeTransmit>,
+        header_name: &str,
+    ) -> &'a HeaderValue {
+        context.request().headers().get(header_name).unwrap()
```

Comment:
Getting rid of the need to `unwrap` because your types are good is such a rusty pleasure.